intake_xarray/base.py

```diff
@@ -52,17 +63,16 @@ def read_partition(self, i):
         if not isinstance(i, (tuple, list)):
             raise TypeError('For Xarray sources, must specify partition as '
                             'tuple')
-        if isinstance(i, list):
-            i = tuple(i)
-        if hasattr(self._ds, 'variables') or i[0] in self._ds.coords:
-            arr = self._ds[i[0]].data
-            i = i[1:]
+        variable, *part = i
+        part = tuple(part)
+        if hasattr(self._ds, 'variables') or variable in self._ds.coords:
+            arr = self._ds[variable].data
         else:
             arr = self._ds.data
         if isinstance(arr, np.ndarray):
-            return arr
```

Comment:
OK so, to be sure, this was the problem - we sent the whole array. I suppose we could have figured out which chunk to send at this point? But maybe indeed better to use Dask's internal logic to do it for us.

Reply:
There might be a good semi-internal dask function to be used here, to extract a block from an array that is already in memory. I think we might be paying a significant tokenization cost by using `dask.array(...)` and should avoid that, but not sure.

Reply:
That's a good point, actually. Dask will assign a token based on the content of the data, which would be slow for a big array, but you can specify the token so that it doesn't do this. I doubt there is a function for quite what is needed here.